services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -149,9 +150,40 @@ export class WorkflowService {
       submitStates: true,
       parentRuntimeDataId: true,
     };
+
     return await this.workflowDefinitionRepository.create({ data, select }, projectIds);
   }
 
+  async cloneWorkflowDefinition(data: WorkflowDefinitionCloneDto, projectIds: TProjectIds) {
+    const select = {
+      reviewMachineId: true,
+      name: true,
+      version: true,
+      projectId: true,
+      isTemplate: true,
+      definitionType: true,
+      definition: true,
+      contextSchema: true,
+      config: true,
+      supportedPlatforms: true,
+      extensions: true,
+      backend: true,
+      persistStates: true,
+      submitStates: true,
+    };
+
+    const workflowDefinition = await this.workflowDefinitionRepository.findTemplateByIdUnscoped(
+      data.id,
+      { select },
+    );
+
+    return await this.workflowDefinitionRepository.create(
+      // @ts-expect-error - types of workflow definition does not propagate to the prisma creation type
+      { data: { ...workflowDefinition, name: data.name }, select },
```

Comment:
`workflowDefinition` has an `id` property, it won't use it?

Reply:
nop, it is not being selected in the select args.